src/rpc/blockchain.cpp

```diff
@@ -79,6 +79,21 @@ double GetNetworkDifficulty(const CBlockIndex* blockindex)
     return GetDifficultyINTERNAL(blockindex, true);
 }
 
+UniValue GetExperimentalFeatures()
+{
+    UniValue experimentalfeatures(UniValue::VARR);
+    if (mapArgs.count("-developerencryptwallet"))
+        experimentalfeatures.push_back("developerencryptwallet");
+    if (mapArgs.count("-developersetpoolsizezero"))
+        experimentalfeatures.push_back("developersetpoolsizezero");
+    if (mapArgs.count("-paymentdisclosure"))
+        experimentalfeatures.push_back("paymentdisclosure");
+    if (mapArgs.count("-insightexplorer"))
+        experimentalfeatures.push_back("insightexplorer");
```

Comment:
Yeah, this is definitely duplicating logic that is currently in `InitExperimentalMode()` in `src/init.cpp`. It's already hard enough to remember all the places we need to update, without this change.

I feel that this would be better as part of a full refactor of experimental features (which grew out organically rather than being planned). Something like the following:

- Move `InitExperimentalMode()` into a new `src/experimental_features.[h,cpp]`
- For each flag, set a global per-feature boolean (stored in the new files) to `true` for each enabled feature, and just check that boolean at each call site, instead of the current `fExperimentalMode && GetBoolArg("-developerencryptwallet", false)`-ish logic we currently use.
- Move this function into the same file, having it test the per-feature global booleans and return a `std::vector<std::string>` of feature names.

Then, adding or removing an experimental feature requires only two changes:
- Modify `src/experimental_features.[h,cpp]` with the globals and option flags
- Modify the feature itself to be gated on the feature's global boolean flag (or remove the gate).

Reply:
Got it, sounds good. Ill give it a try.

Reply:
Refactored following suggested approach at https://github.com/zcash/zcash/pull/4293/commits/d9b135e367eae52cddfb2da03015a6faa6c16638